From 06dc4dc0a26522fb7ea90916a7cebbebe590bb01 Mon Sep 17 00:00:00 2001
Message-Id: <06dc4dc0a26522fb7ea90916a7cebbebe590bb01.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Sun, 25 Jan 2015 15:36:29 +0530
Subject: [PATCH 63/95] MUST_REBASE [VPG]: drivers/video/adf: Add a method to
 destroy dp port

This patch adds a method to clean up dp port related
parameters:
* Destroy the mutex
* Unregister the i2c adapter which was initialized
  during dp_port_init()

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5443
Change-Id: I4db209916846ab81b7fc639a39244b94ec4ad54f
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dp_port.c     |    6 ++++++
 .../video/adf/intel/include/core/vlv/vlv_dp_port.h |    1 +
 2 files changed, 7 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dp_port.c b/drivers/video/adf/intel/core/vlv/vlv_dp_port.c
index c5162c1..c2a01e9 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dp_port.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dp_port.c
@@ -861,6 +861,12 @@ u32 vlv_dp_port_load_panel_delays(struct vlv_dp_port *port)
 	return 0;
 }
 
+void vlv_dp_port_destroy(struct vlv_dp_port *port)
+{
+	mutex_destroy(&port->hw_mutex);
+	i2c_del_adapter(&port->ddc);
+}
+
 bool vlv_dp_port_init(struct vlv_dp_port *port, enum port port_id,
 		enum pipe pipe_id, enum intel_pipe_type type,
 		struct device *dev)
diff --git a/drivers/video/adf/intel/include/core/vlv/vlv_dp_port.h b/drivers/video/adf/intel/include/core/vlv/vlv_dp_port.h
index 2011b4b..474fd85 100644
--- a/drivers/video/adf/intel/include/core/vlv/vlv_dp_port.h
+++ b/drivers/video/adf/intel/include/core/vlv/vlv_dp_port.h
@@ -54,6 +54,7 @@ struct vlv_dp_port {
 
 bool vlv_dp_port_init(struct vlv_dp_port *port, enum port port_id,
 	enum pipe pipe_id, enum intel_pipe_type type, struct device *dev);
+void vlv_dp_port_destroy(struct vlv_dp_port *port);
 
 u32 vlv_dp_port_set_link_pattern(struct vlv_dp_port *port,
 		u8 train_pattern);
-- 
1.7.9.5

